### PR TITLE
S3: Adding a validation method for s3 notification event name

### PR DESCRIPTION
**Pull request by tsugumi-sys**
- Adding a validation method for `S3NotificationEvent`.
- Adding event name validation where necessary.

### CODE REVIEW

**Comment by tsugumi-sys** [comment]
@bblommers 
I think that the CI failures are not related to my changes?

File: moto/s3/notifications.py [hunk 1]
```diff
@@ -89,6 +103,9 @@ def send_event(
     if bucket.notification_configuration is None:
         return
 
+    if not S3NotificationEvent.is_event_valid(event_name):
```
Comment (bblommers) [inline moto/s3/notifications.py:106]: This doesn't seem necessary - we always control when `send_event` is invoked, so there's no point in raising an exception

File: moto/s3/notifications.py [hunk 2]
```diff
@@ -49,11 +51,23 @@ class S3NotificationEvent(str, Enum):
...
+        if event_name in [e[:3] for e in self.events()]:
+            return True
+        return False
+
 
 def _get_s3_event(
     event_name: str, bucket: Any, key: Any, notification_id: str
 ) -> Dict[str, List[Dict[str, Any]]]:
     etag = key.etag.replace('"', "")
+    if not S3NotificationEvent.is_event_valid(event_name):
```
Comment (bblommers) [inline moto/s3/notifications.py:69]: Same here

File: moto/s3/responses.py [hunk 3]
```diff
@@ -2140,7 +2140,7 @@ def _notification_config_from_body(self) -> Dict[str, Any]:
                         n["Event"] = [n["Event"]]
 
                     for event in n["Event"]:
-                        if event not in S3NotificationEvent.events():
+                        if not S3NotificationEvent.is_event_valid(event):
```
Comment (bblommers) [inline moto/s3/responses.py:2143]: This logic is now duplicated into the `Notification`-constructor. That seems to a more logical place to validate this value - so I think we can remove the check here. Or am I missing something?

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/s3/responses.py (hunk 3 above)
Comment (tsugumi-sys) [reply moto/s3/responses.py:2143]: That also sounds good to me :) Thanks

**Review by bblommers — APPROVED** [review]
LGTM - thank you @tsugumi-sys!

**Pull request merged**